demo/core_13_$.utils.code.js

```diff
@@ -137,64 +137,111 @@ $.utils.code.toSourceSafe = function(value) {
   }
 };
 
-$.utils.code.rewriteForEval = function(src) {
+$.utils.code.rewriteForEval = function(src, forceExpression) {
   // Eval treats {} as an empty block (return value undefined).
   // Eval treats {'a': 1} as a syntax error.
   // Eval treats {a: 1} as block with a labeled statement (return value 1).
   // Detect these cases and enclose in parenthesis.
   // But don't mess with: {var x = 1; x + x;}
   // This is consistent with the console on Chrome and Node.
-  try {
-    var ast = $.utils.acorn.parse(src);
-  } catch (e) {
+  // If 'forceExpression' is true, then throw a SyntaxError if the src is
+  // more than one expression (e.g. '1; 2;').
+  var ast = null;
+  if (!forceExpression) {
+    // Try to parse src as a program.
     try {
-      ast = $.utils.acorn.parseExpressionAt(src, 0);
-      if (src.substring(ast.end).trim() !== '') {
-        throw SyntaxError('Not just an expression');
-      }
+      ast = $.utils.acorn.parse(src);
     } catch (e) {
-      // This eval will fail.  Have fun.
-      return src;
+      // ast remains null.
     }
-    // This is an object literal: {'a': 1}
-    return '(' + src + '\n)';
   }
-  if (ast.type === 'Program' && ast.body.length === 1 &&
-      ast.body[0].type === 'BlockStatement') {
-    if (ast.body[0].body.length === 0) {
-      // This is an empty object: {}
-      return '(' + src + '\n)';
+  if (ast) {
+    if (ast.type === 'Program' && ast.body.length === 1 &&
+        ast.body[0].type === 'BlockStatement') {
+      if (ast.body[0].body.length === 0) {
+        // This is an empty object: {}
+        return '({})';
+      }
+      if (ast.body[0].body.length === 1 &&
+          ast.body[0].body[0].type === 'LabeledStatement' &&
+          ast.body[0].body[0].body.type === 'ExpressionStatement') {
+        // This is an unquoted object literal: {a: 1}
+        // There might be a comment, so add a linebreak.
+        return '(' + src + '\n)';
+      }
     }
-    if (ast.body[0].body.length === 1 &&
-        ast.body[0].body[0].type === 'LabeledStatement' &&
-        ast.body[0].body[0].body.type === 'ExpressionStatement') {
-      // This is an object literal: {a: 1}
-      return '(' + src + '\n)';
+    return src;
+  }
+  // Try parsing src as an expression.
+  // This may throw.
+  ast = $.utils.acorn.parseExpressionAt(src, 0);
+  var remainder = src.substring(ast.end).trim();
+  if (remainder !== '') {
+    // Trim off any trailing comments.
```

Comment:
This comment seems out of place, and isn't the most obvious description of what is happening below.

Reply:
Fixed.